File: typescript/17-is-leap-year.ts
// Año bisiesto — Fácil · Números
// Es bisiesto si es divisible por 400, o por 4 pero no por 100.
// Fuente: https://coding-challenges.dev/problems/typescript-ano-bisiesto

export function isLeapYear(year: number): boolean {
  return year % 400 === 0 || (year % 4 === 0 && year % 100 !== 0);
}

File: python/17_is_leap_year.py
"""Año bisiesto — Fácil · Números

Un año es bisiesto si es divisible por 400, o por 4 pero no por 100.

Fuente: https://coding-challenges.dev/problems/python-ano-bisiesto
"""


def is_leap_year(year: int) -> bool:
    return year % 400 == 0 or (year % 4 == 0 and year % 100 != 0)
